activation/poet.go

```diff
@@ -373,7 +383,10 @@ type poetService struct {
 
 	certifier certifierService
 
-	certifierInfoCache cachedData[*certifierInfo]
+	certifierInfoCache cachedData[*types.CertifierInfo]
+	mtx                *sync.Mutex
```

Comment:
You don't need to make `sync.Mutex` a pointer, the zero value can be used directly:

```suggestion
	mtx                sync.Mutex
```

This avoids the need to instantiate the mutex with `mtx = &sync.Mutex{}` See also other usages of `sync.Mutex` in our codebase.